flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -72,15 +76,17 @@ private TaskDeploymentDescriptorFactory(
             JobID jobID,
             boolean allowUnknownPartitions,
             int subtaskIndex,
-            ExecutionEdge[][] inputEdges) {
+            List<ConsumedPartitionGroup> consumedPartitions,
+            Map<IntermediateResultPartitionID, IntermediateResultPartition> resultPartitionsById) {
         this.executionId = executionId;
         this.attemptNumber = attemptNumber;
         this.serializedJobInformation = serializedJobInformation;
         this.taskInfo = taskInfo;
         this.jobID = jobID;
         this.allowUnknownPartitions = allowUnknownPartitions;
         this.subtaskIndex = subtaskIndex;
-        this.inputEdges = inputEdges;
+        this.consumedPartitions = consumedPartitions;
```

Comment:
maybe directly pass in a `List<List<IntermediateResultPartitions>>` in `TaskDeploymentDescriptorFactory#fromExecutionVertex()`? Looks to me this can make things cleaner and then we can drop the `executionGraph.getIntermediateResultPartitionMapping()` method.